core/src/main/scala/cats/Traverse.scala

```diff
@@ -135,12 +135,19 @@ import simulacrum.typeclass
   override def map[A, B](fa: F[A])(f: A => B): F[B] =
     traverse[Id, A, B](fa)(f)
 
+  /**
+   * Akin to [[map]], but allows to keep track of a state value
+   * when calling the function.
+   */
+  def mapAccumulate[A, S, B](fa: F[A], init: S)(f: (S, A) => (S, B)): (S, F[B]) =
```

Comment:
Last nit: I think the order of the parameters should be the same order as the function args: so either `F[A], S, (A, S) => (B, S)` or `S, F[A], (S, A) => (S, B)`

At least IMO.

Reply:
> so either F[A], S, (A, S) => (B, S)

Well, that was argued before and @satorg mentioned that it was common for the state to be on the left like in `foldLeft` _(argument)_ or `State.apply` _(return)_

> or S, F[A], (S, A) => (S, B)

Uhm, that feels weird and would not be consistent with the extension syntax.
Also, this signature is pretty close to `foldLeft` which is `F[A], B, (S, A) => B` which has this _mixed_ order.

Reply:
Also I'm wondering whether should the type parameter order be
   1. `[S, A, B]`
   2. `[A, S, B]`
   3. `[A, B, S]`

I realize it might be annoying (sorry for that), but since it is a sort of design decision, I would like to think twice now rather than stuffing the library with bin-compat hacks later just due to a hasty solution.

Reply:
> Also, I believe, the F[A] is always the first argument in all methods of all typeclases...

Actually, no. I know at least one example where it is not held: [NonEmptyAlternative#prependK](https://github.com/typelevel/cats/blob/3cb763920f177f045e2253be51e52ea2554770ad/core/src/main/scala/cats/NonEmptyAlternative.scala#L37). The syntax for that is defined as usual, though: `xs.prependK(x)`. So such an order is not something absolutely unbearable in Cats.

PS. Actually, it was me who defined `prependK` in that way. But I did it at the request of other reviewers :)